proposals/0012-introduce-reactNativeMetadata.md

```diff
@@ -0,0 +1,329 @@
+---
+title: [DRAFT] Introducing `reactNativeManifest` to `package.json` for React Native specific metadata
+author:
+  - Lorenzo Sciandra
+  - Nicola Corti
+date: 25-01-2021
+---
+
+# RFC0588: [DRAFT] Introducing `reactNativeManifest` to `package.json` for React Native specific metadata
+
+## Summary
+
+> **Warning** ⚠️ This is the initial draft of this proposal. The goal of this draft is to collect feedback and help finalize the spec for this section.
+
+This RFC introduces and formally discuss the concept of _React Native Manifest_, a set of metadata about a React Native project (either app or library). This metadata can be consumed by tooling to enhance the developer experience and support the React Native ecosystem.
+
+Practically, this RFC proposes to add a `reactNativeManifest` section to the `package.json` file of a React Native app or library.
+This new section will allow developers to follow a declarative approach to express capabilities of apps and libraries in a more formalized manner.
+
+## Rationale
+
+The need to specify **capabilities and metadata** for a react-native library is inherent to the React Native ecosystem.
+
+As of today, there is no simple programmatic way to know if a NPM package is a React Native library. Good indicators could be:
+* Presence of a `react-native:*` dependency in the `peerDependencies` section of the `package.json` ([example](https://github.com/RonRadtke/react-native-blob-util/blob/80628d418a5c81439dc2c1a1f05fae6406f0ba7f/package.json#L46C10-L49))
+* Presence of React Native specific files such as `react-native.config.js` files
+Those indicators are all non-exhaustive. 
```

Comment:
How do other libraries determine this? I guess it's not really a problem because we need to be able to catalogue libraries specifically for React Native?

Reply:
> How do other libraries determine this?

What do you mean with "other libraries". Like other ecosystems?

Reply:
Oops yes other ecosystems. I'm guessing they have similar directories. Are they all maintaining it manually?